libs/remix-ui/terminal/src/lib/actions/terminalAction.ts

```diff
@@ -0,0 +1,165 @@
+      // self._appendItem(item)
+    }
+    var scopedCommands = _scopeCommands(append)
+    command(args, scopedCommands, el => append(null, args, blockify(el)))
+  }
+  const help = typeof command.help === 'string' ? command.help : [
+    '// no help available for:', `terminal.command.${name}`
+  ].join('\n')
+  commands[name].toString = () => { return help }
+  commands[name].help = help
+  data.activeFilters.commands[name] = activate && activate.activate
+  if (activate.filterFn) {
+    registerFilter(name, activate.filterFn)
+  }
+  if (name !== ('knownTransaction' || 'unkownTransaction' || 'emptyBlock')) {
+    dispatch({ type: name, payload: { commands: commands, _commands: _commands, data: data } })
+  }
+  const blockify = (el) => {
+    return `<div class="px-4 block_2A0YE0" data-id="block_null">${el}</div>`
+  }
+
+  const _scopeCommands = (append) => {
+    const scopedCommands = {}
+    Object.keys(commands).forEach(function makeScopedCommand (cmd) {
+      var command = _commands[cmd]
+      scopedCommands[cmd] = function _command () {
+        var args = [...arguments]
+        command(args, scopedCommands, el => append(cmd, args, blockify(el)))
+      }
+    })
+    return scopedCommands
+  }
+}
+
+export const filterFnAction = (name, filterFn, dispatch) => {
+  const data: any = {
+    // session: [],
+    // activeFilters: { commands: {}, input: '' },
+    filterFns: {}
+  }
+  data.filterFns[name] = filterFn
+  dispatch({ type: name, payload: { data: data } })
+}
+
+export const registerLogScriptRunnerAction = (event, commandName, commandFn, dispatch) => {
+  event.on('scriptRunner', commandName, (msg) => {
+    commandFn.log.apply(commandFn, msg.data)
+    dispatch({ type: commandName, payload: { commandFn, message: msg.data } })
+  })
+}
+
+export const registerInfoScriptRunnerAction = (event, commandName, commandFn, dispatch) => {
+  event.on('scriptRunner', commandName, (msg) => {
+    commandFn.info.apply(commandFn, msg.data)
+    dispatch({ type: commandName, payload: { commandFn, message: msg.data } })
+  })
+}
+
+export const registerWarnScriptRunnerAction = (event, commandName, commandFn, dispatch) => {
+  event.on('scriptRunner', commandName, (msg) => {
+    commandFn.warn.apply(commandFn, msg.data)
+    dispatch({ type: commandName, payload: { commandFn, message: msg.data } })
+  })
+}
+
+export const registerErrorScriptRunnerAction = (event, commandName, commandFn, dispatch) => {
+  event.on('scriptRunner', commandName, (msg) => {
+    commandFn.error.apply(commandFn, msg.data)
+    dispatch({ type: commandName, payload: { commandFn, message: msg.data } })
+  })
+}
+
+export const registerRemixWelcomeTextAction = (welcomeText, dispatch) => {
+  dispatch({ type: 'welcomeText', payload: { welcomeText } })
+}
+
+export const listenOnNetworkAction = async (props, isListening) => {
+  props.event.trigger('listenOnNetWork', [isListening])
+}
+
+export const initListeningOnNetwork = (props, dispatch) => {
```

Comment:
Rename props to a more relatable name for the function. Also provide argument types if known.